### PR TITLE
Add libtool/2.4.6 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libtool/2.4.6**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by madebr** [comment]
libtool also provides a library `ltdl`.
I have removed it from this recipe to keep this recipe an installer.

Because some packages require this `ltdl` library, I propose creating a new recipe `liblibtool` that provides the headers and library.

File: recipes/libtool/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            self.build_requires("msys2/20190524")
+
+    @contextmanager
+    def _build_context(self):
+        if self.settings.compiler == "Visual Studio":
+            with tools.vcvars(self.settings):
+                with tools.environment_append({"CC": "cl -nologo", "CXX": "cl -nologo",}):
+                    yield
+        else:
+            with tools.no_op():
```
Comment (SSE4) [inline recipes/libtool/all/conanfile.py]: I guess it's not needed, as does nothing, right?

Comment (uilianries) [reply recipes/libtool/all/conanfile.py]: I didn't about [tools.no_op](https://docs.conan.io/en/latest/reference/tools.html#tools-no-op), but it makes sense according the documentation

Comment (madebr) [reply recipes/libtool/all/conanfile.py]: Yup, the `no_op` is a nop here.

**Comment by madebr** [comment]
See #807
If people want `ltdl`, they should require `libtool-ltdl`.
If people want to create a `configure` script, they need to use `libtool`

**Comment by madebr** [comment]
@danimtb Please consider merging this. :relaxed:

**Pull request merged**